app/views/templates/search/search_object.rb

```diff
@@ -155,197 +319,134 @@ def labels
         }
       end
 
-      def data
-        {
-          #agent_pref_label: render_document_show_field_value(document, 'agents.prefLabel'),
-          #agent_begin: render_document_show_field_value(document, 'agents.begin'),
-          #agent_end: render_document_show_field_value(document, 'agents.end'),
-
-          #concepts: render_document_show_field_value(document, 'concepts.prefLabel'),
-
-          #dc_description: render_document_show_field_value(document, 'proxies.dcDescription'),
-          #dc_creator: render_document_show_field_value(document, 'proxies.dcCreator'),
-
-          #dc_format: render_document_show_field_value(document, 'proxies.dcFormat'),
-          #dc_identifier: render_document_show_field_value(document, 'proxies.dcIdentifier'),
-
-          #dc_terms_created: render_document_show_field_value(document, 'proxies.dctermsCreated'),
-          #dc_terms_created_web: render_document_show_field_value(document, 'aggregations.webResources.dctermsCreated'),
-
-          #dc_terms_extent: render_document_show_field_value(document, 'proxies.dctermsExtent'),
-          #dc_title: render_document_show_field_value(document, 'proxies.dcTitle'),
-          #dc_type: render_document_show_field_value(document, 'proxies.dcType'),
-
-          #edm_country: render_document_show_field_value(document, 'europeanaAggregation.edmCountry'),
-          #edm_dataset_name: render_document_show_field_value(document, 'edmDatasetName'),
-          #edm_is_shown_at: render_document_show_field_value(document, 'aggregations.edmIsShownAt'),
-          edm_is_shown_by: render_document_show_field_value(document, 'aggregations.edmIsShownBy'),
-          #edm_language: render_document_show_field_value(document, 'europeanaAggregation.edmLanguage'),
-          #edm_preview: render_document_show_field_value(document, 'europeanaAggregation.edmPreview'),
-          #edm_provider: render_document_show_field_value(document, 'aggregations.edmProvider'),
-          #edm_data_provider: render_document_show_field_value(document, 'aggregations.edmDataProvider'),
-          #edm_rights:  render_document_show_field_value(document, 'aggregations.edmRights'),
-
-          #title: doc_title,
-          #title_extra: doc_title_extra,
-          #type: render_document_show_field_value(document, 'type'),
-
-          #year: render_document_show_field_value(document, 'year')
-        }
-      end
-
       private
 
       def collect_values(fields, doc = document)
-        values = []
-        fields.each { |field| 
-          value = render_document_show_field_value(doc, field)
-          values << value unless value.nil?
-        }
-        values.uniq
+        fields.map do |field|
+          render_document_show_field_value(doc, field)
+        end.compact.uniq
       end
-      
-      
+
       def merge_values(fields, separator = ' ')
         collect_values(fields).join(separator)
       end
 
-      def concept_data
+      def data_section(data)
```

Comment:
Cyclomatic complexity for data_section is too high. [26/6]<br>Assignment Branch Condition size for data_section is too high. [93.99/15]<br>Method has too many lines. [95/10]<br>Perceived complexity for data_section is too high. [29/7]